### ISSUE TITLE
Include examples in the documentation? prepared statements + queries

### ISSUE THREAD

**Issue by jsams** [JuliaDatabases/SQLite.jl#142]
I'm finding the documentation rather difficult to follow. I'm not sure if I'm running into bugs or doing something wrong.

How can I create a prepared statement for a query I intend to run very frequently in a high throughput data computation loop and just swap out parameter values?

But I guess first: how can I write a query that takes parameter values? I've tried this a couple of dozen ways and none seem to work and always with the same error.

```julia
SQLite.query(db, sr"""
               SELECT listener_id, track_id, created_at, author_id
               FROM plays
               WHERE listener_id = ? ORDER BY RANDOM() LIMIT 10;""", [2])
ERROR: AssertionError: you must provide values for all placeholders
```

**Comment by essenciary** [comment]
It's not that... It seems that the documentation is out of sync with the source: https://github.com/JuliaDatabases/SQLite.jl/blob/2610faaaeea7cd92c38c29221101727445f7ab25/src/Source.jl#L107-L115

The right way to do it is 
```
SQLite.query(db, sr"""
               SELECT listener_id, track_id, created_at, author_id
               FROM plays
               WHERE listener_id = ? ORDER BY RANDOM() LIMIT 10;""", values = [2])
```

**Comment by jsams** [comment]
Thanks!

Sorry, not trying to treat this like stack overflow. Putting these here as it really does belong in the documentation imo. I've used sqlite plenty in python and haven't run in to these problems there. Am moving code over to Julia to hopefully get a speed boost. Was expecting the translation to be a little less painful than I'm experiencing, unfortunately.

Still haven't figured out how to keep a compiled version of the sql query around for repeated use in future queries.

How do I pass in a tuple of values for the `IN` sql operator? Tried a few different things. The first query shows that I have a vector of valid tids, but trying to pass them in to queries of the following form runs without error, but returns no results. Manually inserting into the query as `IN (1234, 234, 9872,...)` works as expected.

```julia
    tids = SQLite.query(db, sr"""
        SELECT track_id
        FROM tracks
        ORDER BY RANDOM() LIMIT 10;""")[:track_id]

    SQLite.query(db, sr"""
        SELECT track_id, track_index, author_id
        FROM tracks
        WHERE track_id IN (?);""", values=[tids,])

    SQLite.query(db, sr"""
        SELECT track_id, track_index, author_id
        FROM tracks
        WHERE track_id IN (?);""", values=[tuple(tids...),])
```
